### PR TITLE
fix: blacklist undocumented "--no-" args

### PR DESCRIPTION
**Pull request by Firfi**
**Rationale:**

When `ignoreErrors: false,` is used in YtFlags, although being a valid option,  the program would fail with `youtube-dl: error: no such option: --no-ignore-errors` error.

Indeed, `dargs` without `ignoreFalse: true` would yield an invalid `--no-ignore-errors` argument for yourtube-dl script (the latest version `2021.12.17` at the moment).

**Solution**

Although at least two types of solutions could be implemented, namely whitelist and blacklist, this PR goes the "blacklist" way. The reason being any potential `--no-...` arguments possibly being added in the new youtube-dl versions won't be blocked by youtube-dl-exec code.

The whitelist solution would be more strict and less "update-prone". Please let me know if you'd like the "whitelist" solution better, I'll change the PR accordingly. 

The third way to fix this could be using `ignoreFalse: true` for dargs and adding `no`-prefixed options in the code. It would be a nice modification to the "whitelist" solution.

**Further progress**

More arguments that don't support `--no-` could be easily added to `ignoreFalseFor` which I'll do when the solution used for this issue is agreed upon.

### CODE REVIEW

**Comment by Kikobeats** [comment]
Hello, I don't understand what you are trying to fix; what is the final youtube-dl command you want to cover using this library?

**Comment by Firfi** [comment]
Sorry I wasn't clear in the PR description. For the final youtube-dl command to be `youtube-dl https://myurl`, a user should be able to pass either an empty options object or options object with default values, for example {ignoreErrors: false}`.

However, currently `{ignoreErrors: false}` yields an error, and a not obvious one of that `youtube-dl: error: no such option: --no-ignore-errors`. 

In other words, instead of a correct `youtube-dl https://myurl` it generates an incorrect `youtube-dl --no-ignore-errors https://myurl`.

I also have anticipated that keeping a list of flags won't be an ideal solution and alleviated that by using a "blacklist" (what -`--no` flags aren't allowed) instead of a "whitelist" (what `--no` flags are allowed), which now seems to align with your opinion as well.

Indeed, there is a very small chance they'll add new `--no-` flags that are ignored explicitly here, because for them it'll mean changing a default value for each such flag added, which is unlikely to happen.

I hope this makes sense.

**Comment by Kikobeats** [comment]
Hmmm, I think it's pretty obvious?

`ignoreErrors: false` transformed into `--no-ignore-errors` and youtube-dl CLI doesn't support it.

Why you want to hide an expected the error? why are you passing `ignoreErrors` at first place?

**Comment by Kikobeats** [comment]
According to youtube-dl, the only flag related with ignoring errors is the non ignore error version:

```
    -i, --ignore-errors                  Continue on download errors, for example to skip unavailable videos in a playlist
```

It looks you don't need to pass `ignoreErrors: false` because that's actually the default behavior.

**Comment by Firfi** [comment]
> Hmmm, I think it's pretty obvious?

> why are you passing ignoreErrors at first place?

I'll answer these two questions together because it seems they are likely related. Answering `why are you passing` it's important to determine which "me". A library user who I've been this morning or a library contributor who I'm posing as after reading source code? 🙂 

As a user, the thing I've done is I've ported a python version of config from StackOverflow that happened to contain `"ignoreErrors": False` in it. It has bitten be with `youtube-dl: error: no such option: --no-ignore-errors` which wasn't obvious to a "user-me".

As a contributor, I removed this piece of the config of course, and after reading the source code it's very obvious to me as well why `youtube-dl: error: no such option: --no-ignore-errors` makes total sense. After reading dargs docs etc.

So the question would be, is this a library for users or for contributors, in the sense that they have to engage in the library source code 🙂 

As a user, I expect `{ignoreErrors: false}` to be valid, even if it's a default 🙂 Why? Because it works like that everywhere for (almost?) every language, and it's expected by users! (ref: python lib for the same purpose).

TLDR users will be passing default args, that's what they do.

I hope that makes sense. If this still isn't clear, I will revoke this PR.

**Comment by Kikobeats** [comment]
Thanks for explaining!

Now I can see the point, but I can't merge this. There are a lot of youtube-dl wrappers trying to handle all the user cases to be more user-friendly, being bloated at the end, with a lot of unnecessary code most of the cases can be omitted just spending a moment to understand what should be the right flags.

Plus, youtube-dl is a massive project and often code changes, so I don't really interested into be attached to their changes.

**Comment by Firfi** [comment]
I understand. You are inclined to a more "Linux-way" for this lib and I respect that. Thank you for checking this PR.
